### ISSUE TITLE
Cannot read property 'openConfigFile' of undefined

### ISSUE THREAD

**Issue by IlyaBizyaev** [Microsoft/vscode#38674]
- VSCode Version: Code 1.18.1 (929bacba01ef658b873545e26034d1a8067445e9, 2017-11-16T18:34:22.110Z)
- OS Version: Linux x64 4.4.92-31-default
- Extensions: 

Extension|Author (truncated)|Version
---|---|---
toml|be5|0.0.3
vscode-rust|kal|0.4.2
Go|luk|0.6.67
python|ms-|0.8.0
cpptools|ms-|0.14.2
java|red|0.14.0
vscode-lldb|vad|0.7.1
vscode-java-debug|vsc|0.3.1
debug|web|0.21.2

---
I get this error message when launching debug with any debugger.

Reproduces without extensions: no need in it without extensions.

**Comment by weinand** [comment]
Please provide reproducible steps for a specific debugger extension.

**Comment by IlyaBizyaev** [comment]
For Rust, open any *.rs file, press Debug -> Start debug, choose any debugger. Even works if you just press "More".
Same for C++ and *.cpp files.

**Comment by weinand** [comment]
Please run VS Code without extension `code --disable-extensions` and then create a trivial node program hello.js:
```
console.log("hello world");
```
and start debugging by pressing F5.

Does the error still occur?

**Comment by weinand** [comment]
@isidorn does the error `'openConfigFile' of undefined` ring a bell?

**Comment by IlyaBizyaev** [comment]
Debugging JS seems to work with extensions enabled.

**Comment by isidorn** [comment]
The exception probably comes from here https://github.com/Microsoft/vscode/blob/master/src/vs/workbench/parts/debug/electron-browser/debugService.ts#L722

@IlyaBizyaev so I would speculate that you are doing this from the purple vscode. Is that correct?
Thought `launch` can be null there so I think you incovered a potential null pointer which I can push a fix for.

**Comment by weinand** [comment]
Please try to run VS Code with just one debugger extension enabled, e.g. Rust.
If this works fine, add the next extension. 
I suspect that some extension is causing this.

**Comment by IlyaBizyaev** [comment]
Well.. I'm not sure what "purple" means here, the bottom toolbar is purple.
Here is the log:
```

**Comment by IlyaBizyaev** [comment]
The problem is there for C++ with all extensions disabled.

**Comment by isidorn** [comment]
@IlyaBizyaev I have pushed a fix for this such that a null pointer no longer happens. However your root cause will still be there and that is that your extensions do not seem to support no folder vscode (purple) and are thus failing to start a debug session.
You should open a folder and then start deubugging. You can also open issues in the github repositories of the extensions you are using and let them know they should support no folder workspace debugging

**Comment by ask6155** [comment]
hello,
I'm having this problem while running my C++ code
It's just a  .cpp file

**Comment by IlyaBizyaev** [comment]
Hello ask6155,
Please read the discussion above.

**Comment by ahrherrera** [comment]
Hi @weinand , I am having problems on Linux Mint Serena 18

I followed the instructions in the guide https://docs.microsoft.com/en-us/aspnet/core/tutorials/first-mvc-app-xplat/start-mvc

I installed Code correctly, I ran the `dotnet new mvc ` and I opened the Startup.cs file on Code, I press F5, then a popup request me if switch Node JS or .NET Core. Obviously I switch .Net Core, and appears the message.

Hope this info helps. 

Note: As you said, when I run `code --disable-extensions` and I press F5 to debug It only shows Node JS option.
